src/dts.ts

```diff
@@ -0,0 +1,594 @@
+import * as vscode from 'vscode';
+export { DtsFormatter };
+
+/**
+ * Interface for formatting operation results
+ */
+interface ResultFormat {
+    success: boolean;
+    message?: string;
+}
+
+/**
+ * Provider for DeviceTree diagnostic warnings
+ * Manages diagnostic warnings for DeviceTree files, such as line length issues
+ */
+export class DtsDiagnosticsProvider {
```

Comment:
This file is already quite large. I would recommend to split it up into specific files for DtsDiagnosticsProvider, DtsFormatter and DtsFormatterProvider.